c++/src/Reader.hh

```diff
@@ -96,6 +96,9 @@ namespace orc {
     // For each child of type, select it if one of its children
     // is selected.
     bool selectParents(std::vector<bool>& selectedColumns, const Type& type);
+
+    // For selected map type, select its children, key and value
+    void selectMaps(std::vector<bool>& selectedColumns, const Type& type);
```

Comment:
Does LIST type has same issue?

Reply:
List won't have such issues, because list only has one child. Parent and Child have to be both selected or both unselected.